### PR TITLE
Api for OpenSCAP_foreman

### LINKED ISSUE #7: API
Introduce API for all the actions that can be made on web interface.

### PR DESCRIPTION
**Pull request by shlomizadok**
@isratrade, @ohadlevy please review.
A few concerns:
needed to override a few methods, among them ```resource_name```, ```get_resource``` & ```find_resource``` - this is due the fact that all of openscap models start with ```Scaptimony::model_name```
In the weird case of policy controller I also needed to create a ```policy_url```.

Please review and test.

Fixes #7

### CODE REVIEW

File: app/models/concerns/foreman_openscap/policy_extensions.rb [hunk 1]
```diff
@@ -42,9 +42,11 @@ def hostgroup_ids
     end
 
     def hostgroup_ids=(ids)
+      hostgroup_assets = []
       ids.reject(&:empty?).map do |id|
-        self.assets << assets.where(:assetable_type => 'Hostgroup', :assetable_id => id).first_or_create!
+        hostgroup_assets << assets.where(:assetable_type => 'Hostgroup', :assetable_id => id).first_or_create!
       end
+      self.assets = hostgroup_assets
```
Comment (ohadlevy) [inline app/models/concerns/foreman_openscap/policy_extensions.rb]: why? wont that override existing assets?

File: app/views/api/v2/arf_reports/base.json.rabl [hunk 2]
```diff
@@ -0,0 +1,4 @@
+object @arf_report
+
+attributes :id, :passed, :failed, :othered #:host, :passed, :failed, :othered
```
Comment (ohadlevy) [inline app/views/api/v2/arf_reports/base.json.rabl]: commented out code?

File: app/views/api/v2/arf_reports/base.json.rabl [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+node(:host) { |arf_report| arf_report.host.name }
```
Comment (ohadlevy) [inline app/views/api/v2/arf_reports/base.json.rabl]: why?

File: app/models/concerns/foreman_openscap/policy_extensions.rb (hunk 1 above)
Comment (shlomizadok) [inline app/models/concerns/foreman_openscap/policy_extensions.rb]: This is the default rails behavior. Also, unless this way you could never unset hostgroup(s) from a policy.

File: app/views/api/v2/arf_reports/base.json.rabl (hunk 3 above)
Comment (shlomizadok) [inline app/views/api/v2/arf_reports/base.json.rabl]: So I'll see on the arf_report to which host it ran?

**Comment by shlomizadok** [comment]
Created API, with some functional tests.
To run the tests execute: ```bundle exec rake test:foreman_openscap``` from your foreman directory.
@ohadlevy @isimluk @isratrade please review.
(Good to merge if review / tests okay)

**Comment by shlomizadok** [comment]
@ohadlevy @isimluk ping? Would love to have this reviewed - mainly because of adding test framework to this project, but also cause it enables api

File: config/routes.rb [hunk 4]
```diff
@@ -35,6 +35,9 @@
   namespace :api do
     scope "(:apiv)", :module => :v2, :defaults => {:apiv => 'v2'},
           :apiv => /v1|v2/, :constraints => ApiConstraints.new(:version => 2) do
+      resources :scap_contents
+      resources :policies
```
Comment (isratrade) [inline config/routes.rb]: nikpick, remove  `new` and `edit` actions for API
```ruby
     resources :policies, :except => [:new, :edit]
```

**Comment by isratrade** [comment]
@shlomizadok, a quick look, the code style and API format looks good.

File: config/routes.rb (hunk 4 above)
Comment (shlomizadok) [inline config/routes.rb]: Added

File: app/models/concerns/foreman_openscap/policy_extensions.rb (hunk 1 above)
Comment (shlomizadok) [inline app/models/concerns/foreman_openscap/policy_extensions.rb]: @isimluk - I think this should be amended into 0.3.x, as currently you can't update hostgroups.

**Pull request merged**